core/src/main/scala/kafka/zk/ZkMigrationClient.scala

```diff
@@ -135,20 +138,34 @@ class ZkMigrationClient(zkClient: KafkaZkClient) extends MigrationClient with Lo
   }
 
   def migrateBrokerConfigs(recordConsumer: Consumer[util.List[ApiMessageAndVersion]]): Unit = {
-    val brokerEntities = zkClient.getAllEntitiesWithConfig(ConfigType.Broker)
     val batch = new util.ArrayList[ApiMessageAndVersion]()
+    val zkPasswordEncoder = kafkaConfig.passwordEncoderSecret match {
```

Comment:
Can we pass in a PasswordEncoder to ZkMigrationClient rather than the full KafkaConfig?